### PR TITLE
fix(EVM): Limit max EVM gas by the max burnt gas limit

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Limits to EVM gas to the maximum gas burnt limit per contract to match Wasm execution limit.  

Fixes: https://github.com/near/nearcore/issues/3796

# Test plan:
- A unit test that attaches 300Tgas and spends more than 200Tgas in a EVM function call. The unit test is calling `fib(20)` which spends roughly 240Tgas. 
- CI

### CODE REVIEW

**Review by bowenwang1996 — COMMENTED** [review]
will approve once the unit test is added

**Review by ailisp — COMMENTED** [review]
Change looks good, waiting for unit test. Sorry i didn't anticipate prepaid_gas could be more than max_gas_burnt_limit

**Comment by artob** [comment]
@evgenykuzyakov Here's a quick infinite loop, tested locally to run out of gas:

```solidity
pragma solidity ^0.7.0;

contract Loop {
    function run() public payable returns(uint) {
        uint counter = 0;
        while (true) {
            counter += 1;
        }
        return counter;
    }
}
```

File: nightly/nightly.txt [hunk 1]
```diff
@@ -146,6 +146,9 @@ expensive nearcore test_cases_testnet_rpc test::test_add_access_key_with_allowan
 expensive nearcore test_cases_testnet_rpc test::test_delete_access_key_with_allowance_testnet
 expensive nearcore test_cases_testnet_rpc test::test_access_key_smart_contract_testnet
 expensive nearcore test_cases_testnet_rpc test::test_evm_deploy_call_testnet
+expensive nearcore test_cases_testnet_rpc test::test_evm_fibonacci_gas_limit_testnet
```
Comment (bowenwang1996) [inline nightly/nightly.txt:149]: this file is deprecated. Please add the tests to nightly/tests_for_nayduck.txt. Also are these tests actually very expensive to run?

Comment (evgenykuzyakov) [reply nightly/nightly.txt:149]: No, but sanity CI complains

**Review by ailisp — APPROVED** [review]

File: nightly/nightly.txt (hunk 1 above)
Comment (evgenykuzyakov) [reply nightly/nightly.txt:149]: See https://buildkite.com/nearprotocol/nearcore/builds/5054#b90c35a4-41c6-47ff-b876-9d6f1dad41ae

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**